airflow/hooks/hive_hooks.py

```diff
@@ -396,25 +413,35 @@ def table_exists(self, table_name, db='default'):
 
 class HiveServer2Hook(BaseHook):
     '''
-    Wrapper around the pyhs2 library
+    Wrapper around the impala library
 
-    Note that the default authMechanism is NOSASL, to override it you
+    Note that the default authMechanism is PLAIN, to override it you
     can specify it in the ``extra`` of your connection in the UI as in
-    ``{"authMechanism": "PLAIN"}``. Refer to the pyhs2 for more details.
+    ``{"authMechanism": "NOSASL"}``. Refer to impyla for more details.
     '''
     def __init__(self, hiveserver2_conn_id='hiveserver2_default'):
         self.hiveserver2_conn_id = hiveserver2_conn_id
 
     def get_conn(self):
         db = self.get_connection(self.hiveserver2_conn_id)
-        auth_mechanism = db.extra_dejson.get('authMechanism', 'NOSASL')
+        auth_mechanism = db.extra_dejson.get('authMechanism', 'PLAIN')
```

Comment:
The defaults of impyla differ from pyhs2: as it takes impala as a base for which NOSASL is default while for Hive in a standard setup it is PLAIN. I took hive as default. But yes you should be careful here.